### PR TITLE
Add support for internationalization with TranslateErrors middleware

### PR DESCRIPTION
**Pull request by rschef**
Add support for internationalization with TranslateErrors middleware:

- Rename  HandleChangesetErrors middleware to TranslateErrors
- Implement internationalization using [Gettext](https://github.com/elixir-lang/gettext) and default `Crudry.Translator` module
-  Allow overriding default translator `Crudry.Translator` by adding a custom `translator` in Absinthe's `context`
-  Allow overriding default language `en` by passing a custom `locale` in Absinthe's `context`
-  Translate Ecto error messages using domain defined in `Translator.errors_domain/0`, which defaults to `errors`
-  Translate Ecto schema fields using domain defined in `Translator.schema_fields_domain/0`, which defaults to `schema_fields`
- Add tests
- Improve modules docs

### CODE REVIEW

File: lib/middlewares/translate_errors.ex [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
   ## Usage
 
-  To handle errors for a field, add it after the resolve, using [`middleware/2`](https://hexdocs.pm/absinthe/Absinthe.Middleware.html#module-the-middleware-2-macro):
+  `Cudry.Translator` is used by default to translate error messages to the default locale `en`. You can also use your own Gettext module by adding it and the desired `locale` to to your Absinthe's schema `context/1` function:
 
-      alias Crudry.Middlewares.HandleChangesetErrors
+    def context(context) do
+      context
+      |> Map.put(:translator, MyApp.Translator)
+      |> Map.put(:locale, "pt_BR")
```
Comment (gabrielpra1) [inline lib/middlewares/translate_errors.ex:12]: Wouldn't the default locale already be defined in `MyApp.Translator`?

**Comment by gabrielpra1** [comment]
This is a breaking change, so it should go to https://github.com/gabrielpra1/crudry/tree/v2.0, please rebase with that branch and change the PR to it

File: lib/middlewares/translate_errors.ex (hunk 1 above)
Comment (rschef) [reply lib/middlewares/translate_errors.ex:12]: Yes, indeed. I just wanted to show that the default locale can be dynamically overridden.

Should I remove this line from the docs?

Comment (gabrielpra1) [reply lib/middlewares/translate_errors.ex:12]: Maybe separate this example into 2? One where everything is overriden via the Translator and another where only the locale is overriden

**Comment by rschef** [comment]
Closed in favor of #33

**Pull request closed without merging**